code/game/objects/items/weapons/twohanded.dm

```diff
@@ -1000,20 +1000,29 @@
 		return
 	var/turf/new_item_loc = get_step(current_item_loc, user.dir)
 	var/obj/machinery/disposal/target_bin = locate(/obj/machinery/disposal) in new_item_loc.contents
+	var/obj/structure/janitorialcart/jani_cart = locate(/obj/structure/janitorialcart) in new_item_loc.contents
+	var/obj/vehicle/janicart/jani_vehicle = locate(/obj/vehicle/janicart) in new_item_loc.contents
 	var/trash_amount = 1
 	for(var/obj/item/garbage in current_item_loc.contents)
 		if(!garbage.anchored)
+			if(jani_vehicle?.mybag && garbage.w_class <= WEIGHT_CLASS_SMALL)
+				garbage.forceMove(jani_vehicle.mybag)
+				jani_vehicle.mybag.update_icon_state()
+				to_chat(user, "<span class='notice'>You sweep the pile of garbage into [jani_vehicle].</span>")
```

Comment:
This code is written 3 times now. 2 times exact and once with an `update_icon` instead of `update_icon_state`. Can you make this into a proc?